File: packages/javascript/src/enums.ts
/**
 * Dithering algorithm modes
 * Values match firmware conventions (0-8)
 */
export enum DitherMode {
  /**
   * Direct nearest-color mapping with no error diffusion. Intended for
   * already-quantized graphics only. On limited palettes (especially BWR),
   * continuous-tone photos or large flat mid-tone areas can map to an
   * unexpected ink (e.g. a mid-gray region rendered as solid red); use an
   * error-diffusion mode (e.g. FLOYD_STEINBERG, BURKES) for photographic input.
   */
  NONE = 0,
  BURKES = 1,
  ORDERED = 2,
  FLOYD_STEINBERG = 3,
  ATKINSON = 4,
  STUCKI = 5,
  SIERRA = 6,
  SIERRA_LITE = 7,
  JARVIS_JUDICE_NINKE = 8,
}

File: packages/python/src/epaper_dithering/enums.py
"""Enumeration types for dithering algorithms."""

from __future__ import annotations

from enum import IntEnum


class DitherMode(IntEnum):
    """Image dithering algorithms for e-paper displays.

    Values are compatible with OpenDisplay firmware conventions.
    """

    #: Direct nearest-color mapping with no error diffusion. Intended for
    #: already-quantized graphics only. On limited palettes (especially BWR),
    #: continuous-tone photos or large flat mid-tone areas can map to an
    #: unexpected ink (e.g. a mid-gray region rendered as solid red); use an
    #: error-diffusion mode (e.g. FLOYD_STEINBERG, BURKES) for photographic input.
    NONE = 0
    BURKES = 1
    ORDERED = 2
    FLOYD_STEINBERG = 3
    ATKINSON = 4
    STUCKI = 5
    SIERRA = 6
    SIERRA_LITE = 7
    JARVIS_JUDICE_NINKE = 8
